ql/src/java/org/apache/hadoop/hive/ql/exec/vector/VectorGroupByOperator.java

```diff
@@ -358,6 +360,34 @@ public void close(boolean aborted) throws HiveException {
      */
     private long numRowsCompareHashAggr;
 
+    /**
+     * To track current memory usage.
+     */
+    private long currMemUsed;
+
+    /**
+     * Whether to make use of LRUCache for map aggr buffers or not.
+     */
+    private boolean lruCache;
+
+    class LRUCache extends LinkedHashMap<KeyWrapper, VectorAggregationBufferRow> {
+
+      @Override
+      protected boolean removeEldestEntry(Map.Entry<KeyWrapper, VectorAggregationBufferRow> eldest) {
+        if (currMemUsed > maxHashTblMemory || size() > maxHtEntries || gcCanary.get() == null) {
```

Comment:
this method seems to have been polluted by the "isFull" logic - which is unexpected with this method name
the "isFull" should be moved outside - and remove should only called when the condition is met